java/arcs/core/util/Scheduler.kt

```diff
@@ -83,23 +84,33 @@ class Scheduler(
                 pausedChannel.asFlow().filter { !it }.first()
                 launches.incrementAndGet()
 
-                idlenessChannel.send(false)
                 try {
                     withTimeout(agendaProcessingTimeoutMs) { executeAgenda(agenda) }
                 } finally {
-                    idlenessChannel.send(true)
+                    val agendasLeft = agendasInFlight.getAndDecrement()
```

Comment:
This pattern is repeated twice, here and above with outgoingMessagesInFlight, would be nicer with an extension function, like

```
agendasInFlight.decrementAndRunIfZero {
    idlenessChannel.send(true)
}
```
(or if you can think of a better name)

Reply:
I totally will do this, but I'd prefer to leave it for a later PR.  If I need the pattern again, I'll pop it in a util.